[Chen Yixun] iP

CYX22222003 opened the pull request:
## DukeKorolev

> You can do it quickly, but badly, or you can do it slowly, but well. After a while, everyone will forget that it was fast, but will remember that it was bad. And vice versa.  - Sergei Korolev (<a href="https://en.wikiquote.org/wiki/Sergei_Korolev">source</a>)

DukeKorolev is a ~~robot~~ **chatbot** that help you manage tasks in the daily lives. 
The software is named after the famous Soviet engineer *Sergei Korolev*. The bot is 

- text-based
- user friendly
- EXTRMEMELY FAST 
- free to use 👍 

Some of the features include:

- [X] Create, delete and find task
- [X] Change the status of the event
- [X] Keep track of the date and time of the event

How to use it:

1. download it from <a href=""></a>
2. install jvm
3. run it
4. add your tasks
5. managing them

For Java beginner, you can use it to practice Java. Here is the main method:
```java
public class DukeKorolev {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

CYX22222003 committed:
Rename, greet and exit

CYX22222003 committed:
First implementation of echo

CYX22222003 committed:
Improve echo feature

CYX22222003 committed:
Add list feature

CYX22222003 committed:
Implement tasks class

CYX22222003 committed:
Implement mark features

CYX22222003 committed:
Implement classes of different types of events

CYX22222003 committed:
Finish Implementing different events

CYX22222003 committed:
Modify the implementation of different events

CYX22222003 committed:
Implement simple text ui testing

CYX22222003 committed:
Add new DukeException class

CYX22222003 committed:
Implement error handling mechanism

CYX22222003 committed:
Implement delete function

CYX22222003 committed:
Modify interaction logic
- switch from if-else to switch

CYX22222003 committed:
Correct formatting issues

CYX22222003 committed:
Refactoring
- change the location of parser

CYX22222003 committed:
Implement feature to save list

CYX22222003 committed:
Handle file not existing case

CYX22222003 committed:
Implement loading feature for records of todo and deadline

CYX22222003 committed:
Implement loading features for event

CYX22222003 committed:
Separate storage into a different classes

CYX22222003 committed:
Keep track of korolev.txt

CYX22222003 committed:
Separate UI and main class

CYX22222003 committed:
Separate Command in to a different classes

CYX22222003 committed:
Divide classes into packages

CYX22222003 committed:
Correct linting errors

CYX22222003 committed:
Pull remote gradle branch

CYX22222003 committed:
Merge branch 'add-gradle-support' of https://github.com/CYX22222000/ip

CYX22222003 committed:
Correct formatting issues

CYX22222003 committed:
Update testcase

CYX22222003 committed:
Update testcase 2

CYX22222003 committed:
Update gitignore

CYX22222003 committed:
Update gitignore again

CYX22222003 committed:
Update gitignore again

CYX22222003 committed:
Update gitignore to ignore writing to storage

CYX22222003 committed:
Delete local storage file

CYX22222003 committed:
Merge with add-gradle-support

CYX22222003 committed:
Set up gradle build
- set up gradle build script
- change ui-testing
- add exception handling conditions

CYX22222003 committed:
Finalize gradle configuration

CYX22222003 committed:
Implement the function to parse date for deadline

CYX22222003 committed:
Update event and list class to prepare for further processing

Overload constructor method in KorolevEvent

Remove returned string in commands

CYX22222003 committed:
Implement time parse feature for deadline

CYX22222003 committed:
Implement time parsing feature to handle date and time in event

CYX22222003 committed:
Refactor code to avoid parsing issue

CYX22222003 committed:
Finish implementing the date and time features

CYX22222003 committed:
Implement initial version of test

CYX22222003 committed:
Correct linting error

CYX22222003 committed:
Merge branch 'branch-Level-8'

CYX22222003 committed:
Correct linting error again

CYX22222003 committed:
Merge branch 'branch-Level-7'

CYX22222003 committed:
Add test cases for correct input for DateParser

CYX22222003 committed:
Implement unittest for DateParser

CYX22222003 committed:
Implement unittest for EventParser and KorolevTasks

CYX22222003 committed:
Edit text UI test to adapt to new changes

CYX22222003 committed:
Refactor code to avoid linting error

CYX22222003 committed:
Add JavaDoc comments for methods

CYX22222003 committed:
Add JavaDoc comments for all non-private and classes methods

CYX22222003 committed:
Add more header comments for non-private and classes methods

CYX22222003 committed:
Merge branch 'branch-A-JavaDoc'

CYX22222003 committed:
Correct formatting issues

CYX22222003 committed:
Correct formatting issues in parser

CYX22222003 committed:
Update branch-A-CodingStandard with master

CYX22222003 committed:
Merge branch 'branch-A-CodingStandard'

CYX22222003 committed:
Implement search and find feature

CYX22222003 committed:
Implement search and find feature

CYX22222003 committed:
Correct formatting issues

CYX22222003 committed:
Merge branch 'branch-Level-9'

CYX22222003 committed:
Refactor code to put all files under a package

CYX22222003 committed:
Refactor package name

CYX22222003 committed:
Install packages for GUI

CYX22222003 committed:
Implement the first version of GUI

CYX22222003 committed:
Adjust UI setting of Korolev

CYX22222003 committed:
Adjust UI setting of Korolev again

CYX22222003 committed:
Merge branch 'branch-Level-10'

kaixuan477 commented on gradle/wrapper/gradle-wrapper.properties:
You can use gitignore on the gradle files.

kaixuan477 commented on src/test/java/parser/DateParserTest.java:
```diff
@@ -0,0 +1,93 @@
+import static org.junit.jupiter.api.Assertions.fail;
+public class DateParserTest {
+    @Test
+    public void byStatement_correctFormat_success() throws Exception {
```
Nice naming of test methods!

kaixuan477 commented on src/main/java/korolev/DukeKorolev.java:
```diff
@@ -0,0 +1,14 @@
+public class DukeKorolev {
+    public static void main(String[] args) {
+//        KorolevUI ui = new KorolevUI();
+//        ui.run();
```
You can remove redundant code where no longer needed.

kaixuan477 commented on src/main/java/storage/KorolevStorage.java:
```diff
@@ -0,0 +1,83 @@
+            home, "src", "main", "java", "data");
+
+    private static final Path path = Paths.get(
+            home, "src", "main", "java", "data", "korolev.txt");
```
Static final constants could use SCREAMING_SNAKE_CASE instead of camelCase for variables

kaixuan477 commented on src/main/java/task/KorolevList.java:
```diff
@@ -0,0 +1,220 @@
+    private static String unmarkNotice = "OK, I've marked this task as not done yet:";
+    private static String deleteNotice = "Noted. I've removed this task:";
+
+    private static String filteredNotice = "Here are the matching tasks in your list:";
```
These can be static final constants, see previous comment for constant naming as well.

kaixuan477 commented on src/main/java/ui/DialogBox.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    public static DialogBox getDukeDialog(String text, Image img) {
+        var db = new DialogBox(text, img);
```
Not sure if `var` can be used at this stage, but nice nonetheless.

kaixuan477 reviewed (commented):
Impressive use of OOP and testing, as well as creatively improving the UI of DukeKorolev. Just minor naming standards pointed out, but impressive work nonetheless. 😄

CYX22222003 committed:
Correct linting error

CYX22222003 committed:
Update GUI design

CYX22222003 committed:
Create gradle.yml

CYX22222003 committed:
Update gradle.yml

CYX22222003 committed:
Update CI automation file

CYX22222003 committed:
Add assertions to methods

CYX22222003 committed:
KorolevList class: change modifiers of class variables

There are several class variables with the type of string, which are messages produced by Chatbot.

These variables are not modified throughout the process. Hence, I make them constant to improve the code quality

CYX22222003 committed:
Modify common coding style issues

CYX22222003 committed:
Save new data

CYX22222003 committed:
Merge pull request #1 from CYX22222003/branch-A-Assertions

Add assertions to methods

CYX22222003 committed:
Correct formatting errors

CYX22222003 committed:
Merge branch 'master' into branch-A-CodeQuality

CYX22222003 committed:
Merge pull request #2 from CYX22222003/branch-A-CodeQuality

Modify common coding style issues

CYX22222003 committed:
Add increments to enable tagging feature

CYX22222003 committed:
Add increments to enable display of statistics

CYX22222003 committed:
Update unit test to accomodate new changes

CYX22222003 committed:
Update README: modify the initial draft of user guide

Update the title

Update feature details

CYX22222003 committed:
Update KorolevTask: improve code quality

Correct wrong naming for boolean

CYX22222003 committed:
Update KorolevList: improve code quality

Correct wrong header comments

CYX22222003 committed:
Update KorolevList to improve code quality

CYX22222003 committed:
Update tagging feature.

CYX22222003 committed:
Correct errors in tagging features

CYX22222003 committed:
Correct errors in tagging features again

CYX22222003 committed:
Correct style issues of long methods

CYX22222003 committed:
Merge branch 'master' of https://github.com/CYX22222003/ip

CYX22222003 committed:
Update user manual

CYX22222003 committed:
Update unit test for new tagging system

CYX22222003 committed:
Update unit test for new tagging system

CYX22222003 committed:
Update data file

CYX22222003 committed:
Correct linting errors

CYX22222003 committed:
Correct linting errors

CYX22222003 committed:
Update user manual

CYX22222003 committed:
Update data file and UI demonstration.

CYX22222003 committed:
Update README file

CYX22222003 committed:
Add contents to different parts of README.md

CYX22222003 committed:
Update command summary of README.md

CYX22222003 committed:
Update command summary of README.md and AI.md

CYX22222003 committed:
Update user manual

CYX22222003 committed:
Update user manual and readme.md

CYX22222003 committed:
Trigger new CI action

CYX22222003 committed:
Update link for jar file

CYX22222003 committed:
Enhance UIUX by adjusting GUI parameters

CYX22222003 committed:
Update link for jar file

CYX22222003 committed:
Remove unused methods and variables

CYX22222003 committed:
Update javadoc to correct formatting errors

CYX22222003 committed:
Improve coding standards

CYX22222003 committed:
Update user manual

CYX22222003 committed:
Update EventParser to capture completeness

CYX22222003 committed:
Update user manual

CYX22222003 committed:
Update user manual

CYX22222003 committed:
Update user manual

CYX22222003 committed:
Correct formatting error in user manual

CYX22222003 committed:
Arrange positions of image in user manual

CYX22222003 committed:
Correct minor mistakes in EventParser

CYX22222003 committed:
Update data file

CYX22222003 committed:
Update jar link in user manual

CYX22222003 committed:
Update minor errors in user manual

CYX22222003 committed:
Correct import order

CYX22222003 committed:
Correct other formatting errors

CYX22222003 committed:
Correct other violations of naming conventions

CYX22222003 committed:
Correct other violations of method naming conventions

CYX22222003 committed:
Correct minor formatting errors

CYX22222003 committed:
Add javadocs to public methods

CYX22222003 committed:
Remove magic numbers

CYX22222003 committed:
Update korolev.txt

retrigger CI run

CYX22222003 committed:
Update DateParser.java

trigger PR ci test

CYX22222003 committed:
Merge pull request #5 from CYX22222003/CYX22222003-patch-1

Update DateParser.java
